### ISSUE TITLE
`node-gyp rebuild` doesn't work with clang

### ISSUE THREAD

**Issue by jankeromnes** [chjj/pty.js#149]
When installing `pty.js` on an environment that uses `clang` instead of `gcc`, the `node-gyp rebuild` step fails because two build flags are unsupported.

I successfully worked around this issue by temporarily setting `CC` to `gcc` and `CXX` to `g++` during the `pty.js` install.

Full log:

    :Installing pty.js
    
    > pty.js@0.2.8 install /home/user/.c9/node_modules/pty.js
    > node-gyp rebuild
    
    make: Entering directory `/home/user/.c9/node_modules/pty.js/build'
      CXX(target) Release/obj.target/pty/src/unix/pty.o
    clang: error: unknown argument: '-fno-tree-vrp'
    clang: error: unknown argument: '-fno-tree-sink'
    make: *** [Release/obj.target/pty/src/unix/pty.o] Error 1
    make: Leaving directory `/home/user/.c9/node_modules/pty.js/build'
    gyp ERR! build error 
    gyp ERR! stack Error: `make` failed with exit code: 2
    gyp ERR! stack     at ChildProcess.onExit (/home/user/.c9/node/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:269:23)
    gyp ERR! stack     at ChildProcess.emit (events.js:110:17)
    gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1074:12)
    gyp ERR! System Linux 3.16.0-4-amd64
    gyp ERR! command "node" "/home/user/.c9/node/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
    gyp ERR! cwd /home/user/.c9/node_modules/pty.js
    gyp ERR! node -v v0.12.6
    gyp ERR! node-gyp -v v2.0.1
    gyp ERR! not ok 
    npm ERR! Linux 3.16.0-4-amd64
    npm ERR! argv "/home/user/.c9/node/bin/node" "/home/user/.c9/node/bin/npm" "install" "pty.js@0.2.8"
    npm ERR! node v0.12.6
    npm ERR! npm  v2.11.2
    npm ERR! code ELIFECYCLE
    
    npm ERR! pty.js@0.2.8 install: `node-gyp rebuild`
    npm ERR! Exit status 1
    npm ERR! 
    npm ERR! Failed at the pty.js@0.2.8 install script 'node-gyp rebuild'.
    npm ERR! This is most likely a problem with the pty.js package,
    npm ERR! not with npm itself.
    npm ERR! Tell the author that this fails on your system:
    npm ERR!     node-gyp rebuild
    npm ERR! You can get their info via:
    npm ERR!     npm owner ls pty.js

**Comment by arkrozycki** [comment]
Having similar issue, how did you fix it, i tried setting the env variables for CC and CXX and still getting the same build error.

**Comment by jankeromnes** [comment]
@arkrozycki Doing https://github.com/jankeromnes/dockerfiles/commit/1dd61b271624753b3e3d8c019902020e9f8046b8 fixed the problem for me (the line with `install.sh | bash` does various things including a `pty.js` install, which failed before I added the workaround). Maybe your environment variables are not correctly exported for the `pty.js` build?

**Comment by jankeromnes** [comment]
This now seems to be fixed on `ubuntu 16.04` if you install `clang-3.9` like this:

```
echo "deb http://apt.llvm.org/xenial/ llvm-toolchain-xenial-3.9 main" >> /etc/apt/sources.list
apt-get update && apt-get install clang-3.9
export CC=clang-3.9
export CXX=clang++-3.9
```
